libnymea/coap/coappdu.cpp

```diff
@@ -504,70 +508,91 @@ QByteArray CoapPdu::pack() const
 
 void CoapPdu::unpack(const QByteArray &data)
 {
-    // create a CoapPDU
     if (data.length() < 4) {
         m_error = InvalidPduSizeError;
     }
 
-    quint8 *rawData = (quint8 *)data.data();
-    setVersion((rawData[0] & 0xc0) >> 6);
-    setMessageType(static_cast<MessageType>((rawData[0] & 0x30) >> 4));
-    quint8 tokenLength = (rawData[0] & 0xf);
+    QDataStream stream(data);
+    quint8 flags;
+    stream >> flags;
+
+    setVersion((flags & 0xc0) >> 6);
+//    qCDebug(dcCoap()) << "Version:" << m_version;
+
+    setMessageType(static_cast<MessageType>((flags & 0x30) >> 4));
+//    qCDebug(dcCoap()) << "Message Type:" << messageType();
+
+    quint8 tokenLength = flags & 0x0f;
+//    qCDebug(dcCoap()) << "Token length:" << tokenLength;
 
-    if (tokenLength > 8) {
```

Comment:
I would not remove this one since there is a dedicated error for exactly this case

Reply:
right.. added it back